packages/core/src/agent/AgentModules.ts

```diff
@@ -101,7 +102,9 @@ export type AgentApi<Modules extends ModulesMap> = {
 export type CustomOrDefaultApi<
   CustomModuleType,
   DefaultModuleType extends ApiModule
-> = CustomModuleType extends ApiModule
+> = IsAny<CustomModuleType> extends true
+  ? InstanceType<DefaultModuleType['api']>
```

Comment:
This confuses me a bit.

It seems like we are checking if the CustomModuleType is supplied and if so, we use the `DefaultModuleType['api']`. Should that not be `CustomModuleType['api']` and if not, maybe we should rename that type if we also use it for custom modules.

Small note, I am not very familiar with the generic module type setup so I might have missed something ;).